libpod/runtime_ctr.go

```diff
@@ -505,6 +505,111 @@ func (r *Runtime) removeContainer(ctx context.Context, c *Container, force bool,
 	return cleanupErr
 }
 
+// EvictContainer removes the given container ID. This should be used to remove
+// a container when obtaining a Container struct pointer has failed.
+// Running container will not be stopped.
+// If removeVolume is specified, named volumes used by the container will
+// be removed also if and only if the container is the sole user
+func (r *Runtime) EvictContainer(ctx context.Context, id []byte, removeVolume bool) error {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
+	return r.evictContainer(ctx, id, removeVolume)
+}
+
+// Internal function to evict a container.
+// This does not lock the runtime nor the container.
+// removePod is used only when removing pods. It instructs Podman to ignore
+// infra container protections, and *not* remove from the database (as pod
+// remove will handle that).
+func (r *Runtime) evictContainer(ctx context.Context, id []byte, removeVolume bool) error {
```

Comment:
I think we should probably do a HasContainer() on the ID first - that shouldn't be able to error, even on a potentially invalid configuration. That tells us if it actually exists. If it does, we can proceed with eviction.